src/frameworks/index.ts

```diff
@@ -309,6 +317,12 @@ export async function prepareFrameworks(
     if (publicDir) {
       throw new Error(`hosting.public and hosting.source cannot both be set in firebase.json`);
     }
+    const ssrRegion = (frameworksBackend?.region as string) ?? DEFAULT_REGION;
+    if (!allowedRegionsValues.includes(ssrRegion)) {
+      const notValidRegionError = `Hosting config for site ${site} places server-side content in region ${ssrRegion} which is not known. Valid regions are us-west1, us-central1, us-east1, europe-west1, asia-east1`;
```

Comment:
Map over allowed regions in the message, `ALLOWED_SSR_REGIONS.map(r => r.name).join(', ')`